.github/workflows/osdk_test.yml

```diff
@@ -31,3 +31,19 @@ jobs:
       - name: Unit test
         id: unit_test
         run: cd osdk && cargo build && RUSTUP_HOME=/root/.rustup cargo test
+  
+  # Test OSDK in the same environment as described in the OSDK User Guide in the Asterinas Book.
+  osdk_doc_env_test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    container: asterinas/osdk:0.4.0
+    steps:
+      - run: echo "Running in asterinas/osdk:0.4.0"
+
+      - uses: actions/checkout@v4
+
+      - name: Unit test
+        id: unit_test
+        run: cd osdk && cargo build && RUSTUP_HOME=/root/.rustup cargo test
```

Comment:
Why is it necessary to specify `RUSTUP_HOME` for `cargo test`, but not `cargo build`?

Reply:
Specifing it will ensure that the correct Cargo is used. It's because that Github's checkout action will result in a new user (not root) and thus not using the Rust environment we set up in the container. This only breaks when we invoke cargo in the integration test of OSDK since the OSDK toolchain is not nightly. But I am unsure if this still holds in the new container introduced by @StevenJiang1110

Reply:
This is a good point worth explaining. @StevenJiang1110 Could you add some comments to explain this point?

Reply:
Comments are added to the Dockerfile.